### PR TITLE
Fix broken unwrap bean validation and adapt tests

### PR DESCRIPTION
**Pull request by MelleD**
@tofi86 @wing328 

#49

### CODE REVIEW

File: src/test/java/org/openapitools/jackson/nullable/JsonNullableValueExtractorTest.java [hunk 1]
```diff
@@ -76,7 +75,7 @@ public void testUnwrapList() {
         assertEquals(3, validationResult.size());
         assertTrue(validationResult.stream().anyMatch(c -> c.getPropertyPath().toString().equals("wheels[1].screws") && c.getConstraintDescriptor().getAnnotation().annotationType().getSimpleName().equals("NotNull")));
         assertTrue(validationResult.stream().anyMatch(c -> c.getPropertyPath().toString().equals("wheels[3].screws") && c.getConstraintDescriptor().getAnnotation().annotationType().getSimpleName().equals("NotNull")));
-        assertTrue(validationResult.stream().anyMatch(c -> c.getPropertyPath().toString().equals("persons[0].role") && c.getConstraintDescriptor().getAnnotation().annotationType().getSimpleName().equals("NotNull")));
+        assertTrue(validationResult.stream().anyMatch(c -> c.getPropertyPath().toString().equals("persons[].role") && c.getConstraintDescriptor().getAnnotation().annotationType().getSimpleName().equals("NotNull")));
```
Comment (tofi86) [inline src/test/java/org/openapitools/jackson/nullable/JsonNullableValueExtractorTest.java:78]: @MelleD this however was one of our inital reasons for PR #35: To get the validation to properly return the invalid item in the property path. Why is the item number missing now? Can we fix this and have it return the item number as for the wheels?

**Review by tofi86 — CHANGES_REQUESTED** [review]
Any chance to return the index of the invalid item in the persons collection as it was until now?

File: src/test/java/org/openapitools/jackson/nullable/JsonNullableValueExtractorTest.java (hunk 1 above)
Comment (MelleD) [reply src/test/java/org/openapitools/jackson/nullable/JsonNullableValueExtractorTest.java:78]: No it's normal behavior of bean validation cause it's a Set

Same like:
`  Set<@Valid Person> persons = new HashSet<>();`

It's not related to JsonNullable

**Comment by MelleD** [comment]
> Any chance to return the index of the invalid item in the persons collection as it was until now?

No how i said it's normal behavior of bean validation into a Set

**Comment by tofi86** [comment]
@MelleD Thanks for your quick feedback. I wasn't aware of Set validation not returning the invalid index. I guess I'm fine with that then! :-)

**Review by tofi86 — APPROVED** [review]

**Comment by tofi86** [comment]
At least we got the index for `List` ;-)

**Comment by MelleD** [comment]
> At least we got the index for `List` ;-)

:)

**Comment by wing328** [comment]
Thanks for the PR, which has been merged. Will try to cut a release tomorrow.

**Comment by wing328** [comment]
Released: 
```
[INFO]  * Upload of locally staged artifacts finished.
[INFO]  * Closing staging repository with ID "orgopenapitools-1298".

Waiting for operation to complete...
.........

[INFO] Remote staged 1 repositories, finished with success.
[INFO] Remote staging repositories are being released...

Waiting for operation to complete...
......

[INFO] Remote staging repositories released.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:27 min
[INFO] Finished at: 2023-02-21T14:34:56+08:00
[INFO] ------------------------------------------------------------------------

```

It may take sometime for the new release to be visible in https://search.maven.org

**Pull request merged**